tools/wurf_options.py

```diff
@@ -63,6 +66,20 @@ def resolve(ctx):
 
 class WurfOptions(Options.OptionsContext):
 
+    def execute(self):
+
+        # Execute the context, this will go out and invoke the option(...)
+        # function defined in the wscript. We have some extra tools that
+        # also define the option(...) function to avoid having to call load
+        # in all wscripts to initialize those, we can do it here.
+
+        self.load('wurf_common_tools')
```

Comment:
Are you sure about that last statement? 
If you try to use this version with our existing wscripts, then wurf_common_tools will be loaded twice (in some steps). That might have some unintended side effects. We need to understand the loading mechanism in waf to ascertain those side effects.
I am also trying to make a general point here: "Explicit is better than implicit" ;)